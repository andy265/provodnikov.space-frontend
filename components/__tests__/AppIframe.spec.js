const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
const AppIframe = require('~/components/AppIframe')

describe('AppIframe.vue', () => {
  it('has same HTML structure', () => {
    let wrapper = shallow(AppIframe, {
      propsData: {
        src: 'http://localhost:3000/some_page',
        height: 520,
        width: 300
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('has padding bottom', () => {
    let wrapper = shallow(AppIframe, {
      propsData: {
        src: '',
        height: 9,
        width: 16
      }
    })
    expect(wrapper.vm.paddingBottom).toEqual('56.25%')
  })

  it('and padding bottom <= 100%', () => {
    let wrapper = shallow(AppIframe, {
      propsData: {
        src: '',
        height: 10,
        width: 9
      }
    })
    expect(wrapper.vm.paddingBottom).toEqual('100%')
  })
})
