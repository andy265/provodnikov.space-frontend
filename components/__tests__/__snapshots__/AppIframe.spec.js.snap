// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIframe.vue has same HTML structure 1`] = `
<div class="app-iframe">
    <div class="app-iframe__inner" style="padding-bottom:100%;">
        <iframe src="http://localhost:3000/some_page" allowfullscreen="allowfullscreen" class="app-iframe__iframe"></iframe>
    </div>
</div>
`;
